db/event_handler_test.go

```diff
@@ -25,10 +27,64 @@ func TestSanitizedUrl(t *testing.T) {
+	// Boolean return type handling of CallValidateFunction; Mock up a document change event and
+	// filter function which returns a bool value while calling CallValidateFunction.
+	channels := base.SetFromArray([]string{"Netflix"})
+	docId, body, oldBodyJSON := "doc1", Body{BodyId: "doc1", "key1": "value1"}, ""
+	bodyBytes, _ := base.JSONMarshal(body)
+	event := &DocumentChangeEvent{DocID: docId, DocBytes: bodyBytes, OldDoc: oldBodyJSON, Channels: channels}
+
+	// Boolean return type handling of CallValidateFunction; bool true value.
+	source := `function(doc) { if (doc.key1 == "value1") { return true; } else { return false; } }`
+	filterFunc := NewJSEventFunction(source)
+	result, err := filterFunc.CallValidateFunction(event)
+	assert.True(t, result, "It should return true since doc.key1 is value1")
+	assert.NoError(t, err, "It should return boolean result")
+
+	// Boolean return type handling of CallValidateFunction; bool false value.
+	source = `function(doc) { if (doc.key1 == "value2") { return true; } else { return false; } }`
+	filterFunc = NewJSEventFunction(source)
+	result, err = filterFunc.CallValidateFunction(event)
+	assert.False(t, result, "It should return false since doc.key1 is not value2")
+	assert.NoError(t, err, "It should return boolean result")
+
+	// Parsable boolean string return type handling of CallValidateFunction.
+	source = `function(doc) { if (doc.key1 == "value1") { return "true"; } else { return "false"; } }`
+	filterFunc = NewJSEventFunction(source)
+	result, err = filterFunc.CallValidateFunction(event)
+	assert.True(t, result, "It should return true since doc.key1 is value1")
+	assert.NoError(t, err, "It should return parsable boolean result")
+
+	// Non parsable boolean string return type handling of CallValidateFunction.
+	source = `function(doc) { if (doc.key1 == "value1") { return "TrUe"; } else { return "false"; } }`
+	filterFunc = NewJSEventFunction(source)
+	result, err = filterFunc.CallValidateFunction(event)
+	assert.False(t, result, "It should return false since 'TrUe' is non parsable boolean string")
+	assert.Error(t, err, "It should return parsable throw ParseBool error")
+	assert.Contains(t, err.Error(), `strconv.ParseBool: parsing "TrUe": invalid syntax`)
+
+	// Not boolean and not parsable boolean string return type handling of CallValidateFunction.
+	source = `function(doc) { if (doc.key1 == "Pi") { return 3.14; } else { return 0.0; } }`
+	filterFunc = NewJSEventFunction(source)
+	result, err = filterFunc.CallValidateFunction(event)
+	assert.False(t, result, "It should return not boolean and not parsable boolean string value")
+	assert.Error(t, err, "It should throw Validate function returned non-boolean value error")
+	assert.Contains(t, err.Error(), "Validate function returned non-boolean value.")
+
+	// Simulate CallFunction failure by making syntax error in filter function.
+	source = `func(doc) { if (doc.key1 == "value1") { return true; } else { return false; } }`
```

Comment:
It took me a **LONG** time to figure out what the syntax error was here (`function` -> `func`). Could we make it more obvious?? Something like:

```suggestion
	source = `function(doc) { invalidkeyword if (doc.key1 == "value1") { return true; } else { return false; } }`
```

Reply:
Sure, this is doable.